### ISSUE TITLE
Connect to databases via SSH tunnel

### ISSUE THREAD

**Issue by parsafatehi** [metabase/metabase#1403]
Is it possible to connect to mysql through ssh?

**Comment by evanv** [comment]
I could really use this as well.

**Comment by alepee** [comment]
@parsafatehi you mean create a ssh tunnel to host and connect to db through a localhost port?
```bash
$ ssh -L 27117:127.0.0.1:27017 user@host.domain
```

**Comment by tlrobinson** [comment]
This isn't a problem unique to MySQL, is it?

**Comment by alepee** [comment]
no it's not

**Comment by evanv** [comment]
I'm not sure if that's what @parsafatehi had in mind or not... I would assume so though. And on my end, I can confirm that is exactly what I meant when I initially "me too"'d this issue in Dec @alepee.

**Comment by alepee** [comment]
sorry, I was speaking on my end. As for MySQL, PostgreSQL or Mongo ports can be hidden by server firewall. As I see it, it would make sense to add the possibility to create a SSH tunnel and map ports to access DB through localhost

**Comment by ValentinCreative** [comment]
It would be great something to access your database by SSH. 
Something like Sequel Pro does 
![Sequel Pro SSH](https://www.sononaco.com/wp-content/uploads/2014/02/sequelpro.png)

**Comment by alepee** [comment]
@ValentinCreative yes :smile: 
```bash
#! bin/bash
ssh -L [bind_address:]port:host:host_port
```

**Comment by pretty** [comment]
Would love to see this as well!

**Comment by thehalvo** [comment]
Yeah this would be quite helpful!

**Comment by mishrarohit** [comment]
Have been waiting for this feature to come out so that I can start exploring Metabase. Is this being considered?

**Comment by dfuentes77** [comment]
Yes. We use sequel pro as well and the ssh tunnel feature is very useful. Would be very beneficial to have it here as well.

**Comment by tlrobinson** [comment]
Anyone interested in this feature: how do you need to configure this? password or private key authentication? Is the SSH server generally running on the same machine as the database or do you need to specify both a SSH host+port and the database host+port?

**Comment by dfuentes77** [comment]
A good/basic example is found in Sequel Pro. 

All those options above would allow for more flexibility since everyone has a different security model. But using private key auth with the latter connectivity option is what I'd use.

**Comment by martinjuhasz** [comment]
i think the most common use case is that you don't run metabase on the same machine as your data collection. so yes for host and port on ssh side and on database side. also both password and key auth should be supported if this feature is considered.

**Comment by VictorCano** [comment]
+1

**Comment by MrMauricioLeite** [comment]
+1

**Comment by mdkarp** [comment]
@MrMauricioLeite, @VictorCano - this example worked for me on Heroku: http://stackoverflow.com/questions/21575582/ssh-tunneling-from-heroku

**Comment by gerbz** [comment]
+1

**Comment by mattmonihan** [comment]
+1

**Comment by itmard** [comment]
+1
Can be helpful to connect to servers without direct network access.

**Comment by ArtursO** [comment]
+1

**Comment by yagrossman** [comment]
+100 Having the ability to connect via SSH tunnel using a secure key would be great. A good example of this is Postico

**Comment by rberger** [comment]
Yeah, we could really use this too

**Comment by tlrobinson** [comment]
FYI I have a proof-of-concept in #4005. It's only for Mongo right now and not production-ready but it's a start.

**Comment by pitpit** [comment]
+1

**Comment by sanity** [comment]
+1

As a former Periscope users, Periscope has had this feature for at least 2 years, so to the extent that Metabase is a competitor to Periscope, it would be very useful.

(Sure, you can do it manually with an ssh port forward, but that would be far less convenient)

**Comment by salsakran** [comment]
closed via #4005
